File: Java/String/Simulation/Check If Digits Are Equal in String After Operations I.java
class Solution {
    public boolean hasSameDigits(String s) {
        int[] digits = new int[s.length()];
        for(int i=0;i<s.length();i++){
            digits[i]=s.charAt(i)-'0';
        }
        int len=digits.length;
        while(len>2){
            for(int i=0;i<len-1;i++){
                digits[i]=(digits[i]+digits[i+1])%10;
            }
            len--;
        }
        return digits[0]==digits[1];
    }
}

File: CPP/String/Simulation/Check If Digits Are Equal in String After Operations I.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        vector<int> digits;
        for (auto i : s) {
            digits.push_back(i - '0');
        }
        while (digits.size() > 2) {
            for (int i = 0; i+1 < digits.size(); i++) {
                digits[i] = (digits[i] + digits[i + 1]) % 10;
            }
            digits.pop_back();
        }
        return digits[0] == digits[1];
    }
};
